src/NativeShell/src/cordova/NativeShell.java

```diff
@@ -0,0 +1,188 @@
+                cordova.getActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN);
+            }
+        });
+        PluginResult pluginResult = new PluginResult(PluginResult.Status.OK);
+        callbackContext.sendPluginResult(pluginResult);
+        return true;
+    }
+
+    public boolean disableFullscreen() {
+        cordova.getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                cordova.getActivity().getWindow().getDecorView().setSystemUiVisibility(SYSTEM_UI_FLAG_VISIBLE);
+                cordova.getActivity().getWindow().addFlags(WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN);
+                cordova.getActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
+            }
+        });
+        PluginResult pluginResult = new PluginResult(PluginResult.Status.OK);
+        callbackContext.sendPluginResult(pluginResult);
+        return true;
+    }
+
+    public boolean updateMediaSession() {
+        String action;
+        String itemId;
+        String title;
+        String artist;
+        String album;
+        String imageUrl;
+
+        int duration;
+        int position;
+
+        boolean canSeek;
+        boolean isPaused;
+        boolean isLocalPlayer;
+        try {
+            JSONObject options = args.getJSONObject(0);
+            action = options.getString("action");
+            itemId = options.getString("itemId");
+            title = options.getString("title");
+            artist = options.getString("artist");
+            album = options.getString("album");
+            imageUrl = options.getString("imageUrl");
+
+            duration = options.getInt("duration");
+            position = options.getInt("position");
+
+            canSeek = options.getBoolean("canSeek");
+            isPaused = options.getBoolean("isPaused");
+            isLocalPlayer = options.getBoolean("isLocalPlayer");
+        } catch (Exception e) {
+            callbackContext.error("error: " + e.getMessage());
+            return false;
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            Intent intent = new Intent(cordova.getActivity(), RemotePlayerService.class);
+            intent.setAction(Constants.ACTION_REPORT);
+            intent.putExtra("playerAction", action);
+            intent.putExtra("title", title);
+            intent.putExtra("artist", artist);
+            intent.putExtra("album", album);
+            intent.putExtra("duration", duration);
+            intent.putExtra("position", position);
+            intent.putExtra("imageUrl", imageUrl);
+            intent.putExtra("canSeek", canSeek);
+            intent.putExtra("isPaused", isPaused);
+            intent.putExtra("itemId", itemId);
+            intent.putExtra("isLocalPlayer", isLocalPlayer);
+            cordova.getActivity().startService(intent);
+        }
```

Comment:
hmm, why no `else` here? no playback for <= Lollipop? if so, we should at least make a toast or something... not just report "all is okay" when there's really nothing playing.

Reply:
Playback should work fine for older versions, there just wont be a system notification. We can add support for this in the future, just need to do the work first.